File: Python Word Examples/ShapesApiDemo/addShape.py
import spirecloudword
from spirecloudword import ShapeFormat, Color
from spirecloudword.configuration import *

appId = "your id"
appKey = "your key"
baseUrl = "https://api.e-iceblue.cn"
configuration = Configuration(appId, appKey,baseUrl)
api= spirecloudword.api.ShapesApi(configuration)

file_name = 'Summary of Science.docx'
paragraph_path = 'Section/0/Body/0/Paragraph/2'
shape_properties = ShapeFormat(40, 40, 'Rectangle')
shape_properties._horizontal_origin = 'Page'
shape_properties._vertical_origin = 'Paragraph'
shape_properties._vertical_position = 50
shape_properties._horizontal_position = 150
shape_properties._fill_color = Color(0, 206, 209)
shape_properties._rotation = 45
shape_properties._stroke_weight = 2
shape_properties._stroke_color = Color(173, 255, 47)
shape_properties._text_wrapping_type = 'Both'
shape_properties._text_wrapping_style = 'InFrontOfText'
shape_properties._z_order = 1
extra = {'index_in_paragraph': 0, 'folder': "input", 'dest_file_path': 'output/AddShape.docx'}
result = api.add_shape(file_name, paragraph_path, shape_properties, **extra)

File: JavaScript Word Examples/ShapesApiDemo/addShape.js
(function (){
    var appId = "your id";
    var appKey = "your key";  
    
    var baseUrl = "https://api.e-iceblue.cn";

    var Spirecloudword = require('../../src/index');
    var apiClient = new Spirecloudword.ApiClient(appId, appKey,baseUrl);
    var instance = new Spirecloudword.ShapesApi(apiClient);

    var fileName = "Template.docx";
    var paragraphPath = "Section/0/Body/0/Paragraph/2";
    var tempVar = new Spirecloudword.ShapeFormat();
    tempVar.width = 40;
    tempVar.height = 40;
    tempVar.shapeType = "Rectangle";
    tempVar.horizontalOrigin = "Page";
    tempVar.verticalOrigin = "Paragraph";
    tempVar.verticalPosition = 50.0;
    tempVar.horizontalPosition = 150.0;
    tempVar.fillColor = new Spirecloudword.Color();
    tempVar.fillColor.red = 0;
    tempVar.fillColor.green = 206;
    tempVar.fillColor.blue = 209;
    tempVar.rotation = 45.0;
    tempVar.strokeWeight = 2;
    tempVar.strokeColor = new Spirecloudword.Color();
    tempVar.strokeColor.red = 173;
    tempVar.strokeColor.green = 255;
    tempVar.strokeColor.blue = 47;
    tempVar.textWrappingType = "Both";
    tempVar.textWrappingStyle = "InFrontOfText";
    tempVar.zOrder = 1;
    var shapeProperties = tempVar;
    var opts = { folder: 'input', indexInParagraph: '0' };
    var destFilePath = "output/AddShape_out.docx";
    instance.addShape(fileName, paragraphPath, shapeProperties, destFilePath, opts,
        function (error, data, response) {
            if (error) throw error;
        });
})();